main.tf

```
@@ -81,9 +81,7 @@ resource "google_cloud_scheduler_job" "this" {
   http_target {
     http_method = "POST"
     uri         = "${google_cloud_run_service.this.status[0].url}/http"
-    # TODO implement all payload parameters
-    # https://github.com/sethvargo/gcr-cleaner#payload--parameters
-    body = base64encode(jsonencode(tomap({ "repo" = each.value })))
+    body        = base64encode("{\"allow_tagged\":${each.value.allow_tagged}, \"grace\":\"${each.value.grace}\", \"keep\":${each.value.keep}, \"repos\":[\"${each.value.repo}\"], \"tag_filter\":\"${each.value.tag_filter}\", \"recursive\":${each.value.recursive}}")
```

```suggestion
    body        = base64encode(jsonencode({
      allow_tagged = each.value.allow_tagged,
      grace        = each.value.grace,
      keep         = each.value.keep,
      tag_filter   = each.value.tag_filter,
      recursive    = each.value.recursive,
    }))
```

I think is easier to build the JSON from HCL and not manually :).

---

Unfortunately, this will convert all values into `string` type. However, we need to keep boolean values in boolean type: expl `"allow_tagged": true` and not `"allow_tagged": "true"`

---

I'm septical, the function converts properly the types if I believe the documentation: https://www.terraform.io/docs/language/functions/jsonencode.html.

---

Using what you've suggested
```hcl
    # body        = base64encode("{\"allow_tagged\":${each.value.allow_tagged}, \"grace\":\"${each.value.grace}\", \"keep\":${each.value.keep}, \"repos\":[\"${each.value.repo}\"], \"tag_filter\":\"${each.value.tag_filter}\", \"recursive\":${each.value.recursive}}")
    body        = base64encode(jsonencode({
      allow_tagged = each.value.allow_tagged,
      grace        = each.value.grace,
      keep         = each.value.keep,
      tag_filter   = each.value.tag_filter,
      recursive    = each.value.recursive,
    }))
```
produces this
```console
# ❯ echo eyJhbGxvd190YWdnZWQiOiJmYWxzZSIsImdyYWNlIjoiNzIwaCIsImtlZXAiOiIwIiwicmVjdXJzaXZlIjoiZmFsc2UiLCJ0YWdfZmlsdGVyIjoiIn0= | base64 -d
{"allow_tagged":"false","grace":"720h","keep":"0","recursive":"false","tag_filter":""}
```

---

```suggestion
    body        = base64encode(jsonencode({
      allow_tagged = tobool(each.value.allow_tagged),
      grace        = each.value.grace,
      keep         = each.value.keep,
      tag_filter   = each.value.tag_filter,
      repos        = [each.value.repo],
      recursive    = tobool(each.value.recursive),
    }))
```
That's weird that Terraform converts the bools into string somewhere. But forcing the type before encoding fixes the issue.